### PR TITLE
Set redux carpet

### PR DESCRIPTION
**Pull request by Gdiazdiaz**
## Tasks completed

- Set Redux carpet
- Set books.js with actions for adding and removing.
- Set categories.js with action to display 'under constructions'
- Set configuratioStore.js with combined reducer.

### CODE REVIEW

File: src/redux/categories/categories.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import bookReducer from 'categories/categories';
+
+const rootReducer = Redux.combineReducers({
+  addremo: bookReducer,
+  catredu: categoriesReducer,
+});
+
+const store = Redux.configureStore(rootReducer);
+
+export default store;
```
Comment (Gambit142) [inline src/redux/categories/categories.js]: - [ ] In your `category` slice file, kindly remove the `store` and the `combineReducer` code. You have them set in your _**src/redux/configureStore.js**_ file.

File: src/redux/categories/categories.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { Redux } from '@reduxjs/toolkit';
+import categoriesReducer from 'books/books';
+import bookReducer from 'categories/categories';
```
Comment (Gambit142) [inline src/redux/categories/categories.js]: - [ ] - [ ] You do not need to import this here, kindly remove this import statement. You have imported it in the _**src/redux/configureStore.js**_ file. In redux, it is ideal you configure your store in a file and import all the reducers there. Kindly implement this change. 👍

File: src/redux/categories/categories.js (hunk 1 above)
Comment (Gambit142) [inline src/redux/categories/categories.js]: - [ ] In the categories slice file, the requirements state that you create an _**action**_, an _**action-creator**_ and a _**reducer**_ for categories using the [duck](https://github.com/erikras/ducks-modular-redux) pattern.

I will advise you to execute the following steps:

1. Create the action following the duck pattern. For Example:
`const CHECK_STATUS = <name_of_your_application>/categories/CHECK_STATUS`
2. Create the action creators as functions. The action creators usually return an object that contains the action type and the payload For Example:

```JavaScript
const <name_of_action_creator> = () => {
  type: CHECK_STATUS,
  payload: 'UNDER CONSTRUCTION'
}
```
The action creator should be exported. 👍 💻 
3. Create the reducer which should return **_UNDER CONSTRUCTION_** if the `action.type` is `CHECK_STATUS` or the state if it `action.type` is different. The reducer should be exported as default. 👍 

#### I would advise you to go through this [article](https://www.freecodecamp.org/news/what-is-redux-store-actions-reducers-explained/) to have a good understanding of **_actions_**, **_action creators_** and **_reducers_**. 👍

File: src/redux/books/books.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const ADDBOOK = {
+  type: 'ADD',
+};
```
Comment (Gambit142) [inline src/redux/books/books.js]: - [ ] Your action creators should be a function and it should be exported. Currently, the way you created the action_creator is incorrect. I would advise you to make it a function that takes a `payload` as a parameter and returns an `object` that has `type` and `payload` keys. You can refer to my comment https://github.com/Gdiazdiaz/bookstore/pull/2#discussion_r1085957123 to have an understanding. See the code snippet below to see a sample of an action creator 

image

#### Hint: Your `ADDBOOK` action creator should take the book that it wants to add to the store as a `payload.` 👍

File: src/redux/books/books.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+const REMOVEBOOK = {
+  type: 'REMOVEBOOK',
+};
+
```
Comment (Gambit142) [inline src/redux/books/books.js]: - [ ] Kindly refer to comment https://github.com/Gdiazdiaz/bookstore/pull/2#discussion_r1086007229 for the `REMOVEBOOK` action creator

#### Hint: The `payload` for the `REMOVEBOOK` action creator should be the `id` of the book that will be filtered out when `REMOVEBOOK` action is triggered 💻

File: src/redux/books/books.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+const bookReducer = (state = [], action) => {
+  switch (action.type) {
+    case ADDBOOK:
+      return [...state];
+    case REMOVEBOOK:
+      return state;
+    default:
+      return state;
+  }
+};
```
Comment (Gambit142) [inline src/redux/books/books.js]: - [ ] For the reducer, when the `action.type` is `ADDBOOK`, you should return the `state` and the book which is the `payload`. That is append the payload to the state. `[...state, action.payload]`. When the `action.type` is REMOVEBOOK, you should filter the book with the `id` as `payload` from the state. You can use the `filter` method (preferable) here or any method you can use to remove an item from an array. 👍 💻

File: src/redux/books/books.js (hunk 4 above)
Comment (Gambit142) [inline src/redux/books/books.js]: - [ ] I noticed that you didn't create actions for adding and removing books in the book slice file. Kindly create actions using the [duck](https://github.com/erikras/ducks-modular-redux) pattern. You can also refer to point 1 in my comment https://github.com/Gdiazdiaz/bookstore/pull/2#discussion_r1085957123

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @Gdiazdiaz 👋, 

Good job so far!

## Highlights
- All linters checks passed ✔️ 
- Used `@reduxjs/toolkit` to configure the Redux store ✔️ 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please, remember to tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by t-yanick — APPROVED** [review]
Hi @Gdiazdiaz , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Implemented changes requested by previous reviewer. 
:heavy_check_mark: Descriptive PR

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**